### PR TITLE
Fix for introduce WithCapacityReservation DefinitionStage into the VirtualMachine model

### PR DESCRIPTION
**Pull request by v-hongli1**
- https://github.com/Azure/azure-sdk-for-java/issues/42884

The Capacity Reservation group must contain a Capacity Reservation with matching VM size and zone(s)(if zonal), but no found available VM size and zone(s) in the Azure portal, so disabled Capacity Reservation related test cases.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by weidongxu-microsoft — COMMENTED** [review]
I've some comment on method signature. Let's discuss when Xiaofei's back.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java [hunk 1]
```diff
@@ -1952,6 +1959,17 @@ interface WithUserData {
...
 
+        /** The stage of a virtual machine definition allowing to specify capacity reservation configurations. */
+        interface WithCapacityReservation {
+            /**
+             * Specifies the id of capacity reservation groups for the virtual machine.
+             *
+             * @param capacityReservationGroupsId the id of capacity reservation groups
+             * @return the next stage of the definition
+             */
+            WithCreate withCapacityReservation(String capacityReservationGroupsId);
```
[the remark this answers is not in the document]
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java]: Fixed in the [commit#afffeed](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/afffeedc296ac0cfdb1e68fc173101ed0fa46096)

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java [hunk 2]
```diff
@@ -2626,7 +2656,8 @@ interface Update extends Appliable<VirtualMachine>, Resource.UpdateWithTags<Upda
         UpdateStages.WithBillingProfile, UpdateStages.WithSystemAssignedManagedServiceIdentity,
         UpdateStages.WithUserAssignedManagedServiceIdentity, UpdateStages.WithLicenseType,
         UpdateStages.WithAdditionalCapacities, UpdateStages.WithOSDisk, UpdateStages.WithSecurityFeatures,
-        UpdateStages.WithDeleteOptions, UpdateStages.WithSecurityProfile, UpdateStages.WithUserData {
+        UpdateStages.WithDeleteOptions, UpdateStages.WithSecurityProfile, UpdateStages.WithUserData,
+        UpdateStages.WithCapacityReservation {
```
Comment (dtaveras) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:2660]: Does `copyInnerToUpdateParameter` need to be updated for the update path to correctly set the capacity reservation?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:2660]: Sorry for not noticing the particularity of this service update process. Now fixed it in the [commit#94f34b6](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/94f34b6b7bd61890abc7b590ecd66af4639d78d8)

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:2660]: Sorry for not noticing the particularity of this service logic. Now fixed it in the [commit#b27909e](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/b27909ec61193d4031efb3c408de078cd4b3914e).

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java [hunk 3]
```diff
@@ -1993,6 +1994,11 @@ public String userData() {
         return this.innerModel().userData();
     }
 
+    @Override
+    public String capacityReservationGroupId() {
+        return this.innerModel().capacityReservation().capacityReservationGroup().id();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:1999]: null check

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md [hunk 4]
```diff
@@ -4,11 +4,7 @@
 
 ### Features Added
 
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
+- Supported `capacityReservation` and `withCapacityReservation` methods for `VirtualMachine`.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md:7]: update this

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java [hunk 5]
```diff
@@ -1952,6 +1959,17 @@ interface WithUserData {
...
 
+        /** The stage of a virtual machine definition allowing to specify capacity reservation configurations. */
+        interface WithCapacityReservation {
+            /**
+             * Specifies the id of capacity reservation groups for the virtual machine.
+             *
+             * @param capacityReservationGroupsId the id of capacity reservation groups
+             * @return the next stage of the definition
+             */
+            WithCreate withCapacityReservationGroupsId(String capacityReservationGroupsId);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:1970]: Let us be consistent. I think singular in "capacity reservation group" be more appropriate in these APIs. And getter already be singluar.

```suggestion
            WithCreate withCapacityReservationGroupId(String capacityReservationGroupId);
```

Also update the javadoc to use singular.

@XiaofeiCao Any suggestion about whether for the setter, we use `withCapacityReservationGroup` or `withCapacityReservationGroupId`?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java (hunk 3 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java]: Fixed in the [commit#3a75a4b0](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/3a75a4b0ff711ceaf70d5b6dd3f8c77b69b322ad)

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java (hunk 5 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:1970]: @XiaofeiCao @weidongxu-microsoft 
I think `withCapacityReservationGroup` would be better, and getter can also be changed to `capacityReservationGroup`.

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:1970]: Personally, I think we can use `withCapacityReservationGroup(String capacityReservationGroupId)` for setter. For getter, we should use `capacityReservationGroupId`, to distinguish between a capacity reservation group instance(there's no override for return type).

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java]: Fixed in the [commit#d3df8a2](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/d3df8a26b67b9430da2e4363e5b7c50ae8bfa316)

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md (hunk 4 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md]: Fixed in the [commit#d3df8a2](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/d3df8a26b67b9430da2e4363e5b7c50ae8bfa316)

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java [hunk 6]
```diff
@@ -1993,6 +1994,14 @@ public String userData() {
         return this.innerModel().userData();
     }
 
+    @Override
+    public String capacityReservationGroupId() {
+        if (this.innerModel().capacityReservation() != null) {
+            return this.innerModel().capacityReservation().capacityReservationGroup().id();
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:2000]: also check `this.innerModel().capacityReservation().capacityReservationGroup()`?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java:2001]: Fixed in the [commit#81dafd1](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/81dafd1492a3812074472ab4583c1e13e86b0a76)

**Review by XiaofeiCao — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**